packages/contracts-bedrock/src/L1/SystemConfig.sol

```diff
@@ -144,20 +158,18 @@ contract SystemConfig is OwnableUpgradeable, ISemver {
 
     /// @notice Initializer.
     ///         The resource config must be set before the require check.
-    /// @param _owner             Initial owner of the contract.
-    /// @param _basefeeScalar     Initial basefee scalar value.
-    /// @param _blobbasefeeScalar Initial blobbasefee scalar value.
-    /// @param _batcherHash       Initial batcher hash.
-    /// @param _gasLimit          Initial gas limit.
-    /// @param _unsafeBlockSigner Initial unsafe block signer address.
-    /// @param _config            Initial ResourceConfig.
-    /// @param _batchInbox        Batch inbox address. An identifier for the op-node to find
-    ///                           canonical data.
-    /// @param _addresses         Set of L1 contract addresses. These should be the proxies.
+    /// @param _owner               Initial owner of the contract.
+    /// @param _feeScalars          Initial basefee scalar value.
+    /// @param _batcherHash         Initial batcher hash.
+    /// @param _gasLimit            Initial gas limit.
+    /// @param _unsafeBlockSigner   Initial unsafe block signer address.
+    /// @param _config              Initial ResourceConfig.
+    /// @param _batchInbox          Batch inbox address. An identifier for the op-node to find
+    ///                             canonical data.
+    /// @param _addresses           Set of L1 contract addresses. These should be the proxies.
     function initialize(
         address _owner,
-        uint32 _basefeeScalar,
-        uint32 _blobbasefeeScalar,
+        FeeScalars memory _feeScalars,
```

Comment:
The upgrade will be way easier if we don't have an abi diff in this contract. I recommend we just keep the ABI the same and not set the operator fee values in initialize and instead just make it so that they can be set by the only owner setter. This allows the contract to be upgraded to before the hardfork activates without breaking the L2 due to an unknown upgrade type

Reply:
Ah yes, that simplify a lot of things